Controller/Payment/UpdatePaymentMethod.php

```diff
@@ -0,0 +1,81 @@
+<?php
+declare(strict_types=1);
+
+namespace Magento\Braintree\Controller\Payment;
+
+use Magento\Braintree\Model\Adapter\BraintreeAdapter;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Vault\Api\PaymentTokenManagementInterface;
+
+/**
+ * Class UpdatePaymentMethod
+ */
+class UpdatePaymentMethod extends Action
+{
+    /**
+     * @var BraintreeAdapter
+     */
+    private $adapter;
+    /**
+     * @var PaymentTokenManagementInterface
+     */
+    private $tokenManagement;
+    /**
+     * @var Session
+     */
+    private $session;
+
+    /**
+     * UpdatePaymentMethod constructor.
+     *
+     * @param Context $context
+     * @param BraintreeAdapter $adapter
+     * @param PaymentTokenManagementInterface $tokenManagement
+     * @param Session $session
+     */
+    public function __construct(
+        Context $context,
+        BraintreeAdapter $adapter,
+        PaymentTokenManagementInterface $tokenManagement,
+        Session $session
+    ) {
+        parent::__construct($context);
+        $this->adapter = $adapter;
+        $this->tokenManagement = $tokenManagement;
+        $this->session = $session;
```

Comment:
Please don't instantiate session directly in the constructor and pass customer session proxy with DI instead (constructor should expect SessionManagerInterface
```<argument name="session" xsi:type="object">Magento\Customer\Model\Session\Proxy</argument>```